tasks/Task.cpp

```diff
@@ -0,0 +1,152 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "Task.hpp"
+
+#include <seasocks/PrintfLogger.h>
+#include <seasocks/WebSocket.h>
+#include <seasocks/Server.h>
+
+#include <json/json.h>
+
+#include <string.h>
+#include <iostream>
+
+using namespace controldev_websocket;
+using namespace controldev;
+using namespace seasocks;
+
+struct controldev_websocket::JoystickHandler : WebSocket::Handler {
+    WebSocket *connection = nullptr;
+    Task *task = nullptr;
+    void onConnect(WebSocket *socket) override{
+        if (connection != nullptr){
+            connection->close();
+        }
+        connection = socket;
+    }
+    void onData(WebSocket *socket, const char *data) override{
+        Json::Value msg;
+        if (connection == socket){
+            msg["result"] = Json::Value(task->updateRawCommand(data));
+            if (msg["result"].asBool() && task->controlling){
+                task->raw_cmd_obj.time = base::Time::now();
+                task->_raw_command.write(task->raw_cmd_obj);
+            }
+        } else {
+            msg["result"] = false;
+        }
+        Json::FastWriter fast;
+        connection->send(fast.write(msg));
+    }
+    void onDisconnect(WebSocket *socket) override{
+        if (connection == socket){
+            connection = nullptr;
+        }
+    }
+};
+
+bool Task::updateRawCommand(const char *data){
+    Json::Value jdata;
+    Json::CharReaderBuilder rbuilder;
+    std::unique_ptr<Json::CharReader> const reader(rbuilder.newCharReader());
```

Comment:
Do you have to create it at each request ? I'm guessing it's probably relatively expensive to build ...